packages/headless/src/controllers/core/facets/facet-dependencies-manager/headless-facet-dependencies-manager.ts

```diff
@@ -0,0 +1,194 @@
+import {CoreEngine} from '../../../../app/engine';
+import {facetOptions} from '../../../../app/reducers';
+import {
+  disableFacet,
+  enableFacet,
+} from '../../../../features/facet-options/facet-options-actions';
+import {updateFreezeCurrentValues} from '../../../../features/facets/facet-set/facet-set-actions';
+import {AnyFacetValueRequest} from '../../../../features/facets/generic/interfaces/generic-facet-request';
+import {AnyFacetValue} from '../../../../features/facets/generic/interfaces/generic-facet-response';
+import {
+  CategoryFacetSection,
+  DateFacetSection,
+  FacetOptionsSection,
+  FacetSection,
+  NumericFacetSection,
+} from '../../../../state/state-sections';
+import {loadReducerError} from '../../../../utils/errors';
+
+export interface AnyFacetDependency<T extends AnyFacetValueRequest> {
+  /**
+   * The id of the facet to watch.
+   */
+  parentFacetId: string;
+  /**
+   * The callback that determines whether this dependency is met
+   * @example
+   * ```ts
+   * const fileTypeHasSomeSelectedValue: AnyFacetDependency<FacetValueRequest> = {
+   *   parentFacetId: 'filetype',
+   *   isDependencyMet: (parentValues) => parentValues.some(value => value.state === 'selected')
+   * }
+   * ```
+   * .
+   * @param parentValues - The current values of the facet with the id equal to `parentFacetId`.
+   * @returns Whether the intended dependency is met.
+   */
+  isDependencyMet(parentValues: T[]): boolean;
+}
+
+export interface FacetDependenciesManagerProps {
+  /**
+   * The id of the facet to enable or disable depending on whether dependencies are met.
+   */
+  dependentFacetId: string;
+  /**
+   * The dependencies to watch.
+   *
+   * * If any of these dependencies are met, the dependent facet is enabled.
+   * * If none of these dependencies are met, the dependent facet is disabled.
+   */
+  dependencies: AnyFacetDependency<AnyFacetValueRequest>[];
+}
+
+export interface FacetDependenciesManager {
+  /**
+   * Stops enabling and disabling the dependent facet.
+   */
+  stopWatching(): void;
```

Comment:
maybe `unsubscribe` would be a good way of putting it?

Reply:
I would prefer not `unsubscribe`, since it's a pattern we use for controllers.